File: Agora-RTM2-Tutorial-Android-Java/app/src/main/java/io/agora/rtmdemo/RtmManager.java
package io.agora.rtmdemo;

import io.agora.rtm.RtmClient;
import io.agora.utils.LogUtil;

public class RtmManager {
    private static final RtmManager sInstance = new RtmManager();
    private RtmClient mRtmClient;
    private final EventListener mRtmEventListener = new EventListener();
    private String mAppId;
    private String mToken;

    public static RtmManager the() {
        return sInstance;
    }

    public RtmClient getRtmClient() {
        return mRtmClient;
    }

    public EventListener getRtmEventListener() { return mRtmEventListener; }

    public String getAppId() {
        return mAppId;
    }

    public void setAppId(String appId) {
        mAppId = appId;
    }

    public String getToken() { return mToken; }

    public void setToken(String token) { mToken = token; }

    public void setRtmClient(RtmClient client) {
        mRtmClient = client;
    }

    public void setLogUtil(LogUtil logUtil) {
        mRtmEventListener.setLogUtil(logUtil);
    }
}

File: Agora-RTM2-Tutorial-Android-Kotlin/app/src/main/java/io/agora/rtmdemo/RtmManager.kt
package io.agora.rtmdemo

import io.agora.rtm.RtmClient
import io.agora.utils.LogUtil

class RtmManager {
    var rtmClient: RtmClient? = null
    var appId: String? = null
    var token: String? = null
    val rtmEventListener: EventListener = EventListener()

    companion object {
        private val sInstance = RtmManager()
        fun the(): RtmManager {
            return sInstance
        }
    }

    fun setLogUtil(logUtil: LogUtil?) {
        rtmEventListener.logUtil = logUtil
    }
}